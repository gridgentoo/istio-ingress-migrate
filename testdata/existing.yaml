apiVersion: v1
items:
- apiVersion: extensions/v1beta1
  kind: Ingress
  metadata:
    annotations:
      kubernetes.io/ingress.class: istio
    name: match-cred
    namespace: default
  spec:
    tls:
    - hosts: ["alpha.example.com"]
      secretName: echo-cred
    rules:
    - host: alpha.example.com
      http:
        paths:
        - backend:
            serviceName: echo
            servicePort: 80
          path: /*
          pathType: ImplementationSpecific
- apiVersion: extensions/v1beta1
  kind: Ingress
  metadata:
    annotations:
      kubernetes.io/ingress.class: istio
    name: mismatch-cred
    namespace: default
  spec:
    tls:
    - hosts: ["beta.example.com"]
      secretName: something-else
    rules:
    - host: beta.example.com
      http:
        paths:
        - backend:
            serviceName: echo
            servicePort: 80
          path: /*
          pathType: ImplementationSpecific
- apiVersion: networking.istio.io/v1alpha3
  kind: Gateway
  metadata:
    name: istio-autogenerated-k8s-ingress
    namespace: gke-system
  spec:
    selector:
      istio: ingress-gke-system
    servers:
    - hosts:
      - '*'
      port:
        name: http
        number: 80
        protocol: HTTP2
    - hosts:
      - alpha.example.com
      port:
        name: https
        number: 443
        protocol: HTTPS
      tls:
        credentialName: echo-cred
        mode: SIMPLE
    - hosts:
      - beta.example.com
      port:
        name: https
        number: 443
        protocol: HTTPS
      tls:
        credentialName: echo-cred
        mode: SIMPLE
kind: List
metadata:
  resourceVersion: ""
  selfLink: ""

