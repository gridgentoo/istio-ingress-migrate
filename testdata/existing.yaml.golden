---
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  annotations:
    kubernetes.io/ingress.class: istio
  creationTimestamp: null
  name: match-cred
  namespace: default
spec:
  rules:
  - host: alpha.example.com
    http:
      paths:
      - backend:
          serviceName: echo
          servicePort: 80
        path: /*
        pathType: ImplementationSpecific
  tls:
  - hosts:
    - alpha.example.com
    secretName: echo-cred
status:
  loadBalancer: {}

ERR05: existing TLS settings for Ingress "mismatch-cred" host "beta.example.com" doesn't match expectation. Have "something-else", expected "echo-cred"
---
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  annotations:
    kubernetes.io/ingress.class: istio
  creationTimestamp: null
  name: mismatch-cred
  namespace: default
spec:
  rules:
  - host: beta.example.com
    http:
      paths:
      - backend:
          serviceName: echo
          servicePort: 80
        path: /*
        pathType: ImplementationSpecific
  tls:
  - hosts:
    - beta.example.com
    secretName: echo-cred
status:
  loadBalancer: {}

failures detected during execution
